e2e/tests/init_test.go

```diff
@@ -399,6 +402,53 @@ func (s *suite) TheApplicationReceivesASuccessfulRelayResponseSignedBy(appName s
 	require.Contains(s, stdout, `"result":"0x`)
 }
 
+// TODO_TECHDEBT: Factor out the common logic between this step and waitForTxResultEvent,
+// it is actually not possible since the later is getting the query client from
+// s.scenarioState which seems to cause query client to fail with EOF error.
```

Comment:
```suggestion
// TODO_TECHDEBT: Factor out the common logic between this step and waitForTxResultEvent.
// It is not currently (easily) possible since the latter is getting the query client from
// s.scenarioState, which seems to be the source of the query client's failure.
```